tests/testthat/test_age_model_class.R

```diff
@@ -0,0 +1,127 @@
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0, 0.4), c(0, 0),
+                                       c(0.5, 0.15), c(0, 0))
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0, 0.4), c(0, 0),
+                                       c(0.05, 0.15), c(0, 0))
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0.4), c(0, 0, 0),
+                                       c(0.05, 0.15), c(0, 0))
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0.4), c(0, 0),
+                                       c(0.05, 0, 0.15), c(0, 0))
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0.4), c(0, 0),
+                                       c(0.05, 0.15), c(0, 0, 0))
+    my_model <- `initial_conditions<-`(my_model, 0, c(0, 0), c(0.05, 0.15),
+                                       c(0, 0))
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0.4), 0, c(0.05, 0.15),
+                                       c(0, 0))
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0.4), c(0, 0), 0,
+                                       c(0, 0))
+    my_model <- `initial_conditions<-`(my_model, c(0.4, 0.4), c(0, 0),
+                                       c(0.05, 0.15), 0)
+  })
+})
+
+test_that("can set new transmission parameters for the SEIRAge", {
+  my_model <- new('SEIRAge', name = 'my_model', n_age_categories = 2)
+  my_model <- `transmission_parameters<-`(my_model, 1, 0.5, 0.5)
+  
+  # Test output is correct
+  expected_transpar <- list('b'=1,
+                            'k'=0.5,
+                            'g'=0.5)
+  
+  expect_equal(transmission_parameters(my_model),
+               expected_transpar)
+  
+  # Test input errors
+  expect_error({
+    my_model <- `transmission_parameters<-`(my_model, c(1, 0), 0.5, 0.5)
+    
+    my_model <- `transmission_parameters<-`(my_model, 1, c(1, 0), 0.5)
+    
+    my_model <- `transmission_parameters<-`(my_model, 1, 0.5, c(1, 0))
+  })
+})
+
+test_that("can run simulation for the SEIRAge", {
+  my_model <- new('SEIRAge', name = 'my_model', n_age_categories = 2)
+  my_model <- `initial_conditions<-`(my_model, c(0.6, 0.4), c(0, 0), c(0, 0),
+                                     c(0, 0))
+  
+  my_model <- `transmission_parameters<-`(my_model, 1, 0.5, 0.5)
+
+  expected_output <- data.frame(time=0:2,
+                                S1=rep(0.6,3),
+                                S2=rep(0.4,3),
+                                E1=rep(0,3),
+                                E2=rep(0,3),
+                                I1=rep(0,3),
+                                I2=rep(0,3),
+                                R1=rep(0,3),
+                                R2=rep(0,3))
+  expected_output$Incidence <- cbind(rep(0,3), rep(0,3))
+
+  # Test output is correct
+  expect_equal(simulate_SEIRAge(my_model, seq(0, 2, by = 1)),
+               expected_output)
+
+  # Test input errors
+  expect_error({
+    simulate_SEIRAge(my_model, '0')
+    simulate_SEIRAge(my_model, seq(0, 2, by = 1))})
```

Comment:
What are you trying to test here? This test seems correct to me. It looks like it is giving an 'or' output of all tested statements, you might want an 'and' result.